### PR TITLE
[DEM-282] - Refactor scroll state, navigation and error handling.

### PR DESCRIPTION
**Pull request by vietnogi**
ping @appirio-tech/demandside

### CODE REVIEW

**Comment by vietnogi** [comment]
note, PR is not finish yet.

File: src/client/app/submit-work/about-project/views/name.jade [hunk 1]
```diff
@@ -2,14 +2,22 @@ h2 About your project
...
+    required     = true,
+    ng-model     = "vm.work.name",
+    ng-minlength = "3",
+    ng-pattern   = "/^[0-9a-zA-Z].*$/"
+  )
 
-  ng-messages(for="(vm.next || vm.work.submitAttempted) && vm.validate()" ng-hide="nameForm.name.$pristine && !vm.work.submitAttempted")
-    ng-message(when="required") This field is required
+  ul.invalid-messages
+    li(ng-show="nameForm.name.$dirty && nameForm.name.$error.required") This field is required.
```
Comment (tladendo) [inline src/client/app/submit-work/about-project/views/name.jade]: why not move the `$dirty` logic to the parent element?

Comment (vietnogi) [inline src/client/app/submit-work/about-project/views/name.jade]: good call, i added it to https://github.com/appirio-tech/ap-work-client/pull/145

**Comment by vietnogi** [comment]
merged so product can test

**Pull request merged**